roachtest: fix follower-reads/mixed-version/*

wenyihu6 opened the pull request:
Previously, follower-reads/mixed-version/* tries to change cluster setting from
initFollowerReadsDB which may open a sql connection from a secondary tenenat for
mixed version tests. Since enableClosedTsAutoTune changes SystemVisible cluster
settings, tests would fail with `is only settable by the operator`. This commit
fixes it so that we use system tenant sql connection when changing the cluster
settings.

Fixes: https://github.com/cockroachdb/cockroach/issues/144745
Informs: https://github.com/cockroachdb/cockroach/issues/142194
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/145271)
<!-- Reviewable:end -->

wenyihu6 commented:
Was able to repro this issue by enabling auto-tuning. With this patch: 

```
--- PASS: follower-reads/mixed-version/single-region (5475.11s)
[w0] 2025/04/26 03:12:40 test_runner.go:987: test passed: follower-reads/mixed-version/single-region (run 1)
```

wenyihu6 committed:
roachtest: fix follower-reads/mixed-version/*

Previously, follower-reads/mixed-version/* tries to change cluster setting from
initFollowerReadsDB which may open a sql connection from a secondary tenenat for
mixed version tests. Since enableClosedTsAutoTune changes SystemVisible cluster
settings, tests would fail with `is only settable by the operator`. This commit
fixes it so that we use system tenant sql connection when changing the cluster
settings.

Fixes: https://github.com/cockroachdb/cockroach/issues/144745
Informs: https://github.com/cockroachdb/cockroach/issues/142194
Release note: none

wenyihu6 commented:
TFTR!

bors r=pav-kv
